backend/btrixcloud/crawlconfigs.py

```diff
@@ -1052,6 +1080,94 @@ async def re_add_all_scheduled_cron_jobs(self):
                     flush=True,
                 )
 
+    def _validate_behavior_git_repo(self, repo_url: str, branch: str = ""):
+        """Validate git repository and branch, if specified, exist and are reachable"""
+        git_remote_cmd = ["git", "ls-remote", repo_url, "HEAD"]
+        try:
+            subprocess.run(git_remote_cmd, check=True)
+        # Raises on non-zero exit code (repo doesn't exist or isn't reachable)
+        # pylint: disable=raise-missing-from
+        except subprocess.CalledProcessError:
+            raise HTTPException(
+                status_code=404,
+                detail="custom_behavior_not_found",
+            )
+
+        if branch:
+            time.sleep(2)
+
+            git_remote_cmd = [
+                "git",
+                "ls-remote",
+                "--exit-code",
+                "--heads",
+                repo_url,
+                f"refs/heads/{branch}",
+            ]
+            try:
+                subprocess.run(git_remote_cmd, check=True)
+            # Raises on non-zero exit code (branch ref doesn't exist)
+            # pylint: disable=raise-missing-from
+            except subprocess.CalledProcessError:
+                raise HTTPException(
+                    status_code=404,
+                    detail="custom_behavior_branch_not_found",
+                )
+
+    async def _validate_behavior_url(self, url: str):
+        """Validate behavior file exists at url"""
+        try:
+            timeout = aiohttp.ClientTimeout(total=10)
+            async with aiohttp.ClientSession(timeout=timeout) as session:
+                async with session.get(url) as resp:
+                    if resp.status >= 400:
+                        raise HTTPException(
+                            status_code=404,
+                            detail="custom_behavior_not_found",
+                        )
+        # pylint: disable=raise-missing-from
+        except aiohttp.ClientError:
+            raise HTTPException(
+                status_code=404,
+                detail="custom_behavior_not_found",
+            )
+
+    async def validate_custom_behavior(self, url: str) -> Dict[str, bool]:
+        """Validate custom behavior URL
+
+        Implemented:
+        - Ensure URL is valid (after removing custom git prefix and syntax)
+        - Ensure URL returns status code < 400
+        - Ensure git repository can be reached by git ls-remote and that branch
+        exists, if provided
+        """
+        is_git_repo = False
+        git_branch = ""
+
+        if url.startswith(GIT_PREFIX):
```

Comment:
This seems like some duplication with `_validate_custom_behaviors_url_syntax()`, should it just call that, and maybe return something like `"git", url_parts` to avoid checks in multiple places? A bit of a nitpick